tests/Liuggio/StatsdClient/KeymetricTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Liuggio\StatsdClient;
+
+use Liuggio\StatsdClient\Factory\StatsdDataFactory,
+    Liuggio\StatsdClient\Service\StatsdService;
+
+class KeymetricTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * Test Prefix and suffix key metric
+     */
+    public function testKeyMetric()
+    {
+        $client = $this->getMockBuilder('Liuggio\StatsdClient\StatsdClient')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $factory = new StatsdDataFactory('\Liuggio\StatsdClient\Entity\StatsdData');
```

Comment:
StatsdDataFactory shall be a mock here, your unit testing only StatsdService behavior. Here, make sure that $service->setPrefix(...) calls $mockFactory->setPrefix(...) and that $service->timing calls the $mockFactory->produceStatsdData, that's all.